addons/ai/functions/fnc_waypointParadrop.sqf

```diff
@@ -0,0 +1,63 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Scripted waypoint that makes a group paradrop at the waypoint's position.
+ *
+ * Arguments:
+ * 0: Group <GROUP>
+ * 1: Waypoint Position <ARRAY>
+ *
+ * Return Value:
+ * Waypoint Finished <BOOL>
+ *
+ * Example:
+ * [group, [0, 0, 0]] call zen_ai_fnc_waypointParadrop
+ *
+ * Public: No
+ */
+
+#define MOVE_DELAY 3
+
+params ["_group", "_waypointPosition"];
+
+private _waypoint = [_group, currentWaypoint _group];
+_waypoint setWaypointDescription localize LSTRING(Paradrop);
+
+private _vehicle = vehicle leader _group;
+
+// Exit if the vehicle is not an aircraft
+if !(_vehicle isKindOf "Air") exitWith {true};
+
+private _driver = driver _vehicle;
+private _skill = skill _driver;
+
+// Increase the skill of the pilot for better flying
+_driver setSkill 1;
+_driver allowFleeing 0;
+
+private _nextMove = CBA_missionTime;
+
+waitUntil {
+    // Periodically issue move commands to 2 km past the waypoint's position
+    // Makes the aircraft fly over the waypoint's position and not stop at it
+    if (CBA_missionTime >= _nextMove) then {
+        private _direction = _vehicle getDir _waypointPosition;
+        private _position = _waypointPosition getPos [2000, _direction];
+        _vehicle move _position;
+
+        _nextMove = CBA_missionTime + MOVE_DELAY;
+    };
+
+    sleep 0.5;
+
+    // Check if the aircraft is close enough to the waypoint to begin paradropping
+    // Distance is based on the current speed of aircraft and the number of passengers
+    private _passengers = {assignedVehicleRole _x select 0 == "cargo"} count crew _vehicle;
+    _vehicle distance2D _waypointPosition < vectorMagnitude velocity _vehicle * 1.5 * _passengers / 2 max 50
```

Comment:
1) I think it should be `1.2` instead of `1.5`, since `ZEN_fnc_ejectPassengers` has 1.2 s as delay between the passengers.
If you want to account for the 1.5 s before opening the chute, you could add  `_vehicleSpeed * 1.5` in addition.
Anyway, we can improve the trajectories later on by doing displacement measurements like I did for HALO once.
2) Also consider measuring the distance between `_position` and `_vehicle` rather than `_waypointPosition` and `_vehicle`. That way you don't need the set the minimum to 50 m.

Reply:
1. Can try 1.2 instead of 1.5.
2. `_position` is 2 km past `_waypointPosition` in the direction from the aircraft to `_waypointPosition`. This would cause the units to eject late and not over the waypoint's position.

Reply:
I mean something like
```suggestion
_vehicle distance2D _position <  (vectorMagnitude velocity _vehicle) * (1.2 * _passengers / 2 + 1.5) + 2000;
```
The offsets we use are in respect to the direction of travel, which is better described by the vector `_position vectorDiff getPos _vehicle` than  `_waypointPosition vectorDiff getPos _vehicle`. Hence, you will get better results in the case the aircraft does not exactly pass the target location.

Reply:
Also, the numbers should be put into preprocessor defines as per the no magic numbers rule.